railties/lib/rails/tasks/annotations.rake

```diff
@@ -2,21 +2,21 @@
 
 require "rails/source_annotation_extractor"
 
-desc "Enumerate all annotations (use notes:optimize, :fixme, :todo for focus)"
 task :notes do
-  Rails::SourceAnnotationExtractor.enumerate "OPTIMIZE|FIXME|TODO", tag: true
+  Rails::SourceAnnotationExtractor::Annotation.notes_task_deprecation_warning
+  Rails::Command.invoke :notes
 end
 
 namespace :notes do
   ["OPTIMIZE", "FIXME", "TODO"].each do |annotation|
-    # desc "Enumerate all #{annotation} annotations"
     task annotation.downcase.intern do
-      Rails::SourceAnnotationExtractor.enumerate annotation
+      Rails::SourceAnnotationExtractor::Annotation.notes_task_deprecation_warning
+      Rails::Command.invoke :notes, ["--annotations", annotation]
```

Comment:
In the future we'd want to the command infrastructure to support: `Rails::Command.invoke :notes, annotations: annotation`

Reverting back to a CLI-like argument passing is ridiculous when we're already in Ruby 😄

Reply:
Agreed. I'll look into adding support for it in a follow up PR. Any reasons why it doesn't support that already?

Reply:
We haven't had a lot of command invocations from the Ruby side internally in Rails yet, so I just never got to it.

But when we eventually want to expose commands to apps, running them from other commands in a nicer way is the way to go.